core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -618,10 +618,8 @@ class ReplicaManager(val config: KafkaConfig,
     quota.isThrottled(topicPartition) && quota.isQuotaExceeded && !isReplicaInSync
   }
 
-  def getMagicAndTimestampType(topicPartition: TopicPartition): Option[(Byte, TimestampType)] =
-    getReplica(topicPartition).flatMap { replica =>
-      replica.log.map(log => (log.config.messageFormatVersion.messageFormatVersion, log.config.messageTimestampType))
-    }
+  def getMagic(topicPartition: TopicPartition): Option[(Byte)] =
```

Comment:
That's actually not a tuple (otherwise we'd have failing tests), but the parenthesis is not necessary, my bad. Fixed.